common/sgx/report.c

```diff
@@ -91,7 +91,18 @@ oe_result_t oe_parse_report(
         OE_RAISE(OE_INVALID_PARAMETER);
 
     if (header->report_size + sizeof(oe_report_header_t) != report_size)
-        OE_RAISE(OE_INCORRECT_REPORT_SIZE);
+        OE_RAISE_MSG(
+            OE_INCORRECT_REPORT_SIZE,
+            "Report size is invalid. "
+            "Header report size: %d bytes, report buffer size: %d",
+            header->report_size,
+            report_size);
+
+    if (header->report_size < sizeof(sgx_report_t))
```

Comment:
This check need to be done after checking report type, to be either sgx_report_t or sgx_quote_t (plus its signature length)